Mathlib/FieldTheory/Adjoin.lean

```diff
@@ -581,19 +590,27 @@ theorem adjoin_simple_toSubalgebra_of_integral (hα : IsIntegral F α) :
   rwa [isAlgebraic_iff_isIntegral]
 #align intermediate_field.adjoin_simple_to_subalgebra_of_integral IntermediateField.adjoin_simple_toSubalgebra_of_integral
 
+/-- Characterize `IsSplittingField` with `IntermediateField.adjoin` instead of `Algebra.adjoin`. -/
+theorem _root_.isSplittingField_iff_intermediateField {p : F[X]} :
+    p.IsSplittingField F E ↔ p.Splits (algebraMap F E) ∧ adjoin F (p.rootSet E) = ⊤ := by
+  rw [← toSubalgebra_injective.eq_iff,
+      adjoin_algebraic_toSubalgebra fun _ ↦ isAlgebraic_of_mem_rootSet]
+  exact ⟨fun ⟨spl, adj⟩ ↦ ⟨spl, adj⟩, fun ⟨spl, adj⟩ ↦ ⟨spl, adj⟩⟩
+
+-- Note: p.Splits (algebraMap F E) also works
 theorem isSplittingField_iff {p : F[X]} {K : IntermediateField F E} :
     p.IsSplittingField F K ↔ p.Splits (algebraMap F K) ∧ K = adjoin F (p.rootSet E) := by
   suffices _ → (Algebra.adjoin F (p.rootSet K) = ⊤ ↔ K = adjoin F (p.rootSet E)) by
```

Comment:
Is it possible to use your new lemma to golf this proof?

Reply:
It doesn't quite work because [Polynomial.adjoin_rootSet_eq_range](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Polynomial/Splits.html#Polynomial.adjoin_rootSet_eq_range) is about Subalgebras not IntermediateFields. I wonder whether we should change IsSplittingField to use IntermediateField.adjoin ...

Reply:
This can be a good idea, but let's keep it for another PR.